src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -46,13 +47,24 @@ class FilterButton extends React.Component<Props, State> {
         const initialCondition = this.createCondition();
 
         this.setState({
-            conditions: [initialCondition],
+            transientConditions: [initialCondition],
         });
     }
 
+    componentDidUpdate() {
+        const { conditions } = this.props;
+        const { transientConditions } = this.state;
+        if (transientConditions.length === 0 && conditions.length !== transientConditions.length) {
```

Comment:
The only time we should update transientConditions is if there's a mismatch between ```this.props.conditions``` and ```this.state.transientConditions```.

Reply:
The empty array check is due to the ```onClose``` method setting transientConditions to be an empty array when the FilterButton dropdown is closed.

Reply:
>         if (transientConditions.length === 0 && conditions.length !== transientConditions.length) {

Could this be expressed differently e.g. "if conditions is not empty and transient conditions is empty..." 

I'd like to see "const wasFlyoutOpen = transientConditions.length === null". That logic is implied and I'd like to see it made more obvious.

The if-condition should say "if flyout was not open but is open now, then copy to transient conditions". Is this possible?